src/fidesops/service/connectors/query_config.py

```diff
@@ -121,17 +131,56 @@ def update_value_map(self, row: Row, policy: Policy) -> Dict[str, Any]:
         for a given customer_id will be replaced with null values.
 
         """
-        rule_to_collection_fields = self.build_rule_target_fields(policy)
+        rule_to_collection_fields: Dict[
+            Rule, List[str]
+        ] = self.build_rule_target_fields(policy)
 
         value_map: Dict[str, Any] = {}
         for rule, field_names in rule_to_collection_fields.items():
             strategy_config = rule.masking_strategy
-            strategy = get_strategy(
+            if not strategy_config:
+                continue
+            strategy: MaskingStrategy = get_strategy(
                 strategy_config["strategy"], strategy_config["configuration"]
             )
 
             for field_name in field_names:
-                value_map[field_name] = strategy.mask(row[field_name])
+                masking_override: MaskingOverride = [
+                    MaskingOverride(field.data_type, field.length)
+                    for field in self.node.node.collection.fields
+                    if field.name == field_name
+                ][0]
+                not_null_masking: bool = strategy_config.get("strategy") != NULL_REWRITE
+                if not_null_masking:
+                    if not masking_override.data_type:
+                        logger.warning(
+                            f"Unable to generate a query for field {field_name} due to: data_type required on fields for the {strategy_config['strategy']} masking strategy"
+                        )
+                        continue
+                    if not strategy.data_type_supported(
+                        data_type=masking_override.data_type.name
+                    ):
+                        logger.warning(
+                            f"Unable to generate a query for field {field_name} due to: data_type of {masking_override.data_type} is not supported for the {strategy_config['strategy']} masking strategy"
+                        )
+                        continue
+                val: Any = row[field_name]
+                masked_val = strategy.mask(val)
+                logger.info(
+                    f"Generated the following masked val for field {field_name}: {masked_val}"
```

Comment:
I think this should be debug: info log on each mask will generate _a lot_ of log messages.